internal/tagging/aws.go

```diff
@@ -1,14 +1,43 @@
 package tagging
 
-import "github.com/env0/terratag/internal/convert"
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/env0/terratag/internal/convert"
+	"github.com/env0/terratag/internal/tag_keys"
+	"github.com/hashicorp/hcl/v2/hclwrite"
+)
 
 func tagAutoscalingGroup(args TagBlockArgs) (*Result, error) {
-	// for now, we count on it that if there's a single "tag" in the schema (unlike "tags" block),
-	// then no "tags" interpolation is used, but rather multiple instances of a "tag" block
 	// https://www.terraform.io/docs/providers/aws/r/autoscaling_group.html
-	if err := convert.AppendTagBlocks(args.Block, args.Tags); err != nil {
+
+	var tagsMap map[string]string
+	if err := json.Unmarshal([]byte(args.Tags), &tagsMap); err != nil {
 		return nil, err
 	}
 
+	tagsAttr := args.Block.Body().GetAttribute("tags")
+	if tagsAttr != nil {
```

Comment:
Note:
this fix will not work for 0.11 - assuming this is okay since 0.11 was released 5 years ago.